src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -0,0 +1,295 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\ArgumentResolver;
+
+use Doctrine\DBAL\Types\ConversionException;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\NoResultException;
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\Persistence\ObjectManager;
+use Symfony\Bridge\Doctrine\Attribute\Entity;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Yields the entity matching the criteria provided in the route.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class EntityValueResolver implements ArgumentValueResolverInterface
+{
+    private array $defaultOptions;
+
+    public function __construct(
+        private ManagerRegistry $registry,
+        private ?ExpressionLanguage $language = null,
+        array $defaultOptions = []
+    ) {
+        $this->defaultOptions = array_merge([
+            'entity_manager' => null,
+            'expr' => null,
+            'mapping' => [],
+            'exclude' => [],
+            'strip_null' => false,
+            'id' => null,
+            'evict_cache' => false,
+            'auto_mapping' => true,
+            'attribute_only' => false,
+        ], $defaultOptions);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(Request $request, ArgumentMetadata $argument): bool
```

Comment:
How does this `supports` method behave in a project where we have multiple instances of the resolver registered (one for the ORM and one for the ODM) ?

Reply:
The methods relies on `ManagerRegistry::getManager` from `\Doctrine\Persistence`. 
Looking at its code, when several managers are registered it picks the first one that is able to load metadata for the provided class (in our case, the Argument's class we are resolving).

We use the same behavior in `Symfony\Form\DoctrineType`.

Reply:
The call to `$this->getManager(...)` should return null for one or the other, and this `support` method will return false.